core/src/main/java/org/mskcc/cbio/portal/servlet/GetProfileDataJSON.java

```diff
@@ -169,10 +172,21 @@ protected void doPost(HttpServletRequest httpServletRequest,
                 //Get raw data (plain text) for each profile
                 for (String geneticProfileId: geneticProfileIds) {
                     try {
-                        ArrayList<String> tmpProfileDataArr = GeneticAlterationUtil.getGeneticAlterationDataRow(
-                                gene,
-                                internalSampleIds,
-                                DaoGeneticProfile.getGeneticProfileByStableId(geneticProfileId));
+                    	ArrayList<String> tmpProfileDataArr;
+                    	GeneticProfile geneticProfile = DaoGeneticProfile.getGeneticProfileByStableId(geneticProfileId);
+                    	if (geneticProfile.getGeneticAlterationType().equals(GeneticAlterationType.GENESET_SCORE)) {
```

Comment:
I think that the assumption here was that geneticProfileIds is always of length 1, but the code does not guarantee this, so yes...it is a bit hacky. There is a type for entities, see `/core/src/main/java/org/mskcc/cbio/portal/model/EntityType.java` (actually used here as well, see EntityType.GENESET). Perhaps we can use this. So the web-service would need an extra entity type argument and fetch the data based on that rather than trying to infer via the genetic profile.